From 41bfbeef9ca275c7906d4e4399564460fad4b072 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:33 -0400
Subject: [ethernet] i40e: Some FD message fixes

Message-id: <1409658042-22689-273-git-send-email-sassmann@redhat.com>
Patchwork-id: 90475
O-Subject: [RHEL7.1 PATCH 272/281] i40e: Some FD message fixes
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

Change the message that gets printed when adding/deleting a filter to
the SB, so that user can tell if a filter was added or deleted.
Print filter add failures only in case of SB filters. For ATR the
information is not useful to the user and hence suppress it unless in
higher debug mode.

Change-ID: I78d7a7a6ecfa82a38a582b0d7b4da038355e3735
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit f7233c5491156bc530ed965d4fabc55db35f2e37)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_txrx.c b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
index a6cc8d6..267c0e6 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_txrx.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_txrx.c
@@ -228,9 +228,14 @@ static int i40e_add_del_fdir_udpv4(struct i40e_vsi *vsi,
     fd_data->pctype, fd_data->fd_id, ret);
   err = true;
  } else {
-  dev_info(&pf->pdev->dev,
-    "PCTYPE:%d, Filter send OK for fd_id:%d\n",
-    fd_data->pctype, fd_data->fd_id);
+  if (add)
+   dev_info(&pf->pdev->dev,
+     "Filter OK for PCTYPE %d loc = %d\n",
+     fd_data->pctype, fd_data->fd_id);
+  else
+   dev_info(&pf->pdev->dev,
+     "Filter deleted for PCTYPE %d loc = %d\n",
+     fd_data->pctype, fd_data->fd_id);
  }
  return err ? -EOPNOTSUPP : 0;
 }
@@ -290,9 +295,13 @@ static int i40e_add_del_fdir_tcpv4(struct i40e_vsi *vsi,
     fd_data->pctype, fd_data->fd_id, ret);
   err = true;
  } else {
-  dev_info(&pf->pdev->dev,
-    "PCTYPE:%d, Filter send OK for fd_id:%d\n",
-    fd_data->pctype, fd_data->fd_id);
+  if (add)
+   dev_info(&pf->pdev->dev, "Filter OK for PCTYPE %d loc = %d)\n",
+     fd_data->pctype, fd_data->fd_id);
+  else
+   dev_info(&pf->pdev->dev,
+     "Filter deleted for PCTYPE %d loc = %d\n",
+     fd_data->pctype, fd_data->fd_id);
  }
 
  return err ? -EOPNOTSUPP : 0;
@@ -359,9 +368,14 @@ static int i40e_add_del_fdir_ipv4(struct i40e_vsi *vsi,
      fd_data->pctype, fd_data->fd_id, ret);
    err = true;
   } else {
-   dev_info(&pf->pdev->dev,
-     "PCTYPE:%d, Filter send OK for fd_id:%d\n",
-     fd_data->pctype, fd_data->fd_id);
+   if (add)
+    dev_info(&pf->pdev->dev,
+      "Filter OK for PCTYPE %d loc = %d\n",
+      fd_data->pctype, fd_data->fd_id);
+   else
+    dev_info(&pf->pdev->dev,
+      "Filter deleted for PCTYPE %d loc = %d\n",
+      fd_data->pctype, fd_data->fd_id);
   }
  }
 
@@ -443,6 +457,10 @@ static void i40e_fd_handle_status(struct i40e_ring *rx_ring,
   I40E_RX_PROG_STATUS_DESC_QW1_ERROR_SHIFT;
 
  if (error == (0x1 << I40E_RX_PROG_STATUS_DESC_FD_TBL_FULL_SHIFT)) {
+  if ((rx_desc->wb.qword0.hi_dword.fd_id != 0) ||
+      (I40E_DEBUG_FD & pf->hw.debug_mask))
+   dev_warn(&pdev->dev, "ntuple filter loc = %d, could not be added\n",
+     rx_desc->wb.qword0.hi_dword.fd_id);
 
   /* filter programming failed most likely due to table full */
   fcnt_prog = i40e_get_cur_guaranteed_fd_count(pf);
@@ -470,7 +488,7 @@ static void i40e_fd_handle_status(struct i40e_ring *rx_ring,
    }
   } else {
    dev_info(&pdev->dev,
-     "FD filter programming failed due to incorrect filter parameters\n");
+    "FD filter programming failed due to incorrect filter parameters\n");
   }
  } else if (error ==
      (0x1 << I40E_RX_PROG_STATUS_DESC_NO_FD_ENTRY_SHIFT)) {
-- 
1.7.1